src/core/tiledmesh/qgscesiumtilesdataprovider.cpp

```diff
@@ -45,58 +47,88 @@
 
 QgsCesiumTilesDataProviderSharedData::QgsCesiumTilesDataProviderSharedData() = default;
 
-void QgsCesiumTilesDataProviderSharedData::setTilesetContent( const QString &tileset )
+void QgsCesiumTilesDataProviderSharedData::setTilesetContent( const QString &tileset, const QgsCoordinateTransformContext &transformContext )
 {
   mTileset = json::parse( tileset.toStdString() );
 
   // parse root
   {
     const auto &root = mTileset[ "root" ];
     // parse root bounding volume
+
+    // TODO -- read crs from metadata tags. Need to find real world examples of this. And can metadata crs override
+    // the EPSG:4979 requirement from a region bounding volume??
+
     {
       const auto &rootBoundingVolume = root[ "boundingVolume" ];
       if ( rootBoundingVolume.contains( "region" ) )
       {
         const QgsBox3d rootRegion = QgsCesiumUtils::parseRegion( rootBoundingVolume[ "region" ] );
         if ( !rootRegion.isNull() )
         {
+          mBoundingVolume = std::make_unique< QgsTiledMeshNodeBoundingVolumeRegion >( rootRegion );
           mZRange = QgsDoubleRange( rootRegion.zMinimum(), rootRegion.zMaximum() );
-          // The latitude and longitude values are given in radians!
-          // TODO -- is this ALWAYS the case? What if there's a region root bounding volume, but a transform object present? What if there's crs metadata specifying a different crs?
-          mCrs = QgsCoordinateReferenceSystem( QStringLiteral( "EPSG:4979" ) );
-          const double xMin = rootRegion.xMinimum() * 180 / M_PI;
-          const double xMax = rootRegion.xMaximum() * 180 / M_PI;
-          const double yMin = rootRegion.yMinimum() * 180 / M_PI;
-          const double yMax = rootRegion.yMaximum() * 180 / M_PI;
-          mExtent = QgsRectangle( xMin, yMin, xMax, yMax );
+          mLayerCrs = QgsCoordinateReferenceSystem( QStringLiteral( "EPSG:4979" ) );
+          mMeshCrs = QgsCoordinateReferenceSystem( QStringLiteral( "EPSG:4979" ) );
```

Comment:
Hmm... here mesh will be still in EPSG:4978 though - it's just the bounding volume region being in EPSG:4979...